### PR TITLE
Fix doc and clippy lints (2024-W04) (ENG-2278)

### PR DESCRIPTION
**Pull request by nickgerace**
## Description

Ran the following two commands to fix warnings found:

- `cargo clippy --all-targets --all-features`
- `cargo doc --all --all-features --document-private-items`

## Additional Note

I wonder why our `prelude-si` checks aren't including the `cargo doc` equivalent checks, in particular. Something to investigate...

### CODE REVIEW

**Comment by nickgerace** [comment]
bors merge

File: lib/cyclone-server/src/tower.rs [hunk 1]
```diff
@@ -1,5 +1,3 @@
-pub use limit_requests::{LimitRequest, LimitRequestLayer};
```
Comment (fnichol) [inline lib/cyclone-server/src/tower.rs:1]: Why are these type re-exports removed?

File: lib/pinga-server/src/config.rs [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
 use si_crypto::{CryptoConfig, SymmetricCryptoServiceConfig, SymmetricCryptoServiceConfigFile};
 use si_data_nats::NatsConfig;
 use si_data_pg::PgPoolConfig;
 use si_std::CanonicalFileError;
+use std::{env, path::Path};
 use telemetry::prelude::*;
 use thiserror::Error;
+use ulid::Ulid;
 
-pub use si_crypto::CycloneKeyPair;
```
Comment (fnichol) [inline lib/pinga-server/src/config.rs:13]: Why is this export removed?

**Review by fnichol — COMMENTED** [review]
I'm wondering about a few of the type exports that were removed? Was this an oversight or on purpose?

**Comment by fnichol** [comment]
@nickgerace The reason these checks aren't yet in CI was for 2 reasons:

1. There were a *lot* of failures that needed to be addressed
2. The current Buck2 rule relies on running `cargo doc` which all take the global Cargo lock at made running these pretty untenable. Theses days, from what I recall, I think I'd be able to make the Cargo-free version of these rules but until then it was going to be far more painful on the runtime of `si/review-pr` to be worth turning them on.

**Comment by nickgerace** [comment]
bors cancel

**Comment by nickgerace** [comment]
Need to review your comments @fnichol! Cancelled the merge.

File: lib/pinga-server/src/config.rs (hunk 2 above)
Comment (nickgerace) [reply lib/pinga-server/src/config.rs:13]: Clippy seems to believe the import is unused. Does not mean that we don't want to keep though, but I see I've seen it on `main` for a couple weeks now.
```
warning: unused import: `si_crypto::CycloneKeyPair
```

File: lib/cyclone-server/src/tower.rs (hunk 1 above)
Comment (nickgerace) [reply lib/cyclone-server/src/tower.rs:1]: Clippy seems to believe the import is unused. Does not mean that we don't want to keep though, but I see I've seen it on `main` for a couple weeks now.
```
warning: unused import: limit_requests::{LimitRequest, LimitRequestLayer}
```

**Comment by nickgerace** [comment]
@fnichol good to know both of those reasons! I run into these lints pretty often, so found some down time and decided to hand comb through them. As far as the re-exports go, `clippy` might be being overly enthusiastic. I replied to your comments accordingly.

**Comment by nickgerace** [comment]
Now's not the time and there's increasing drift with main.

**Pull request closed without merging**